{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Alineamiento con Biopython"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## de la pagina\n",
    "https://medium.com/towards-data-science/pairwise-sequence-alignment-using-biopython-d1a9d0ba861f\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import pairwise2, SeqIO\n",
    "from Bio.pairwise2 import format_alignment\n",
    "from Bio import AlignIO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "X = \"ACGGGT\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "Y = \"ACG\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "alignements = pairwise2.align.globalxx(X, Y)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ACGGGT\n",
      "||||||\n",
      "AC--G-\n",
      "  Score=3\n",
      "\n",
      "ACGGGT\n",
      "||||||\n",
      "AC-G--\n",
      "  Score=3\n",
      "\n",
      "ACGGGT\n",
      "||||||\n",
      "ACG---\n",
      "  Score=3\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for a in alignements:\n",
    "    print (format_alignment(*a))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ACGGGT\n",
      "|||||\n",
      "AC--G-\n",
      "  Score=3\n",
      "\n",
      "ACGGGT\n",
      "||||\n",
      "AC-G--\n",
      "  Score=3\n",
      "\n",
      "ACGGGT\n",
      "|||\n",
      "ACG---\n",
      "  Score=3\n",
      "\n"
     ]
    }
   ],
   "source": [
    "alignements = pairwise2.align.localxx(X, Y)\n",
    "for a in alignements:\n",
    "    print (format_alignment(*a))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ACGGGT\n",
      "||||||\n",
      "ACG---\n",
      "  Score=5.3\n",
      "\n"
     ]
    }
   ],
   "source": [
    "alignements =  pairwise2.align.globalms(X, Y, 2, -1, -0.5, -0.1)\n",
    "for a in alignements:\n",
    "    print (format_alignment(*a))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# example 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ACGGGT\n",
      "||||||\n",
      "AC--G-\n",
      "  Score=3\n",
      "\n",
      "ACGGGT\n",
      "||||||\n",
      "AC-G--\n",
      "  Score=3\n",
      "\n",
      "ACGGGT\n",
      "||||||\n",
      "ACG---\n",
      "  Score=3\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Import pairwise2 module\n",
    "from Bio import pairwise2\n",
    "\n",
    "# Import format_alignment method\n",
    "from Bio.pairwise2 import format_alignment\n",
    "\n",
    "# Define two sequences to be aligned\n",
    "X = \"ACGGGT\"\n",
    "Y = \"ACG\"\n",
    "\n",
    "# Get a list of the global alignments between the two sequences ACGGGT and ACG\n",
    "# No parameters. Identical characters have score of 1, else 0.\n",
    "# No gap penalties.\n",
    "alignments = pairwise2.align.globalxx(X, Y)\n",
    "\n",
    "# Use format_alignment method to format the alignments in the list\n",
    "for a in alignments:\n",
    "    print(format_alignment(*a))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# example 2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ACGGGT\n",
      "|||||\n",
      "AC--G-\n",
      "  Score=3\n",
      "\n",
      "ACGGGT\n",
      "||||\n",
      "AC-G--\n",
      "  Score=3\n",
      "\n",
      "ACGGGT\n",
      "|||\n",
      "ACG---\n",
      "  Score=3\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Import pairwise2 module\n",
    "from Bio import pairwise2\n",
    "\n",
    "# Import format_alignment method\n",
    "from Bio.pairwise2 import format_alignment\n",
    "\n",
    "# Define two sequences to be aligned\n",
    "X = \"ACGGGT\"\n",
    "Y = \"ACG\"\n",
    "\n",
    "# Get a list of the local alignments between the two sequences ACGGGT and ACG\n",
    "# No parameters. Identical characters have score of 1, else 0.\n",
    "# No gap penalties.\n",
    "alignments = pairwise2.align.localxx(X, Y)\n",
    "\n",
    "# Use format_alignment method to format the alignments in the list\n",
    "for a in alignments:\n",
    "    print(format_alignment(*a))\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# example 3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ACGGGT\n",
      "||||||\n",
      "ACG---\n",
      "  Score=5.3\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Import pairwise2 module\n",
    "from Bio import pairwise2\n",
    "\n",
    "# Import format_alignment method\n",
    "from Bio.pairwise2 import format_alignment\n",
    "\n",
    "# Define two sequences to be aligned\n",
    "X = \"ACGGGT\"\n",
    "Y = \"ACG\"\n",
    "\n",
    "# Get a list of the global alignments between the two sequences ACGGGT and ACG satisfying the given scoring\n",
    "# A match score is the score of identical chars, else mismatch score.\n",
    "# Same open and extend gap penalties for both sequences.\n",
    "alignments = pairwise2.align.globalms(X, Y, 2, -1, -0.5, -0.1)\n",
    "\n",
    "# Use format_alignment method to format the alignments in the list\n",
    "for a in alignments:\n",
    "    print(format_alignment(*a))\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Ejercicio"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/home/user1/Desktop/jupyter'"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/user1/Desktop/data/fasta\n"
     ]
    }
   ],
   "source": [
    "cd ~/Desktop/data/fasta/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio.SeqRecord import SeqRecord"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Nav1_2_B_halotolerans \t Nav1\n",
      "Nav1_8_B_aquimaris \t Nav1\n",
      "Nav2_2_A_mediterranea \t Nav2\n",
      "PET1_2_H_chejuensis \t PET1\n",
      "PET1_8_P_oceani \t PET1\n",
      "PET2_2_A_mediterranea \t PET2\n",
      "PET2_8_A_mediterranea \t PET2\n",
      "PETsn1_2_A_mediterranea \t PETsn1\n",
      "PETsn1_8_A_mediterranea \t PETsn1\n",
      "PHBb1_2_L_janthinus \t PHBb1\n"
     ]
    }
   ],
   "source": [
    "f= open(\"secuencias_anotadas.fasta\")\n",
    "n=0\n",
    "secuencias=[]\n",
    "for rec in SeqIO.parse(f, \"fasta\"):\n",
    "    rec1 = rec.id[:rec.id.find(\"_\")]\n",
    "    #rec.id = rec1\n",
    "    print (rec.id, \"\\t\", rec1)\n",
    "    linea =SeqRecord((rec.seq), id=rec1, description=rec.description)\n",
    "    secuencias.append(linea)\n",
    "\n",
    "\n",
    "    "
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "f= open(\"secuencias.fasta\")\n",
    "n=0\n",
    "secuencias=[]\n",
    "for rec in SeqIO.parse(f, \"fasta\"):\n",
    "    print (rec.id, rec.description)"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "SeqIO.write(secuencias, \"secuencias_alinear.fasta\", \"fasta\")\n"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "f= open(\"secuencias_anotadas.fasta\")\n",
    "n=0\n",
    "secuencias=[]\n",
    "for rec in SeqIO.parse(f, \"fasta\"):\n",
    "\n",
    "    if rec.id[-1]==\"R\":\n",
    "        linea =SeqRecord((rec.seq.reverse_complement()), id=rec.id[:-1]+\"RC\", description=rec.description+\" RC\")\n",
    "    else:\n",
    "        linea =SeqRecord((rec.seq), id=rec.id, description=rec.description)\n",
    "    secuencias.append(linea)\n",
    "    print (linea.id)\n",
    "    "
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "SeqIO.write(secuencias, \"../fasta/secuencias_alinear.fasta\", \"fasta\")\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Alineamiento"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "from Bio.Align.Applications import ClustalwCommandline"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "cline = ClustalwCommandline(\"clustalw2\", infile=\"secuencias_comp.fas\")\n",
    "print(cline)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "ename": "ApplicationError",
     "evalue": "Non-zero return code 126 from '/home/user1/Applications/clustalw/clustalw2 -infile=secuencias_anotadas.fasta', message '/bin/sh: 1: /home/user1/Applications/clustalw/clustalw2: Permission denied'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m\u001b[0m",
      "\u001b[0;31mApplicationError\u001b[0mTraceback (most recent call last)",
      "\u001b[0;32m<ipython-input-9-daf128ed278b>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      2\u001b[0m \u001b[0mclustalw_cline\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mClustalwCommandline\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mclustalw_exe\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0minfile\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m\"secuencias_anotadas.fasta\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      3\u001b[0m \u001b[0;32massert\u001b[0m \u001b[0mos\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpath\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0misfile\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mclustalw_exe\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m\"Clustal W executable missing\"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 4\u001b[0;31m \u001b[0mstdout\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstderr\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mclustalw_cline\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m/home/jupyter/miniconda3/lib/python3.6/site-packages/Bio/Application/__init__.py\u001b[0m in \u001b[0;36m__call__\u001b[0;34m(self, stdin, stdout, stderr, cwd, env)\u001b[0m\n\u001b[1;32m    521\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mreturn_code\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    522\u001b[0m             raise ApplicationError(return_code, str(self),\n\u001b[0;32m--> 523\u001b[0;31m                                    stdout_str, stderr_str)\n\u001b[0m\u001b[1;32m    524\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mstdout_str\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstderr_str\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    525\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mApplicationError\u001b[0m: Non-zero return code 126 from '/home/user1/Applications/clustalw/clustalw2 -infile=secuencias_anotadas.fasta', message '/bin/sh: 1: /home/user1/Applications/clustalw/clustalw2: Permission denied'"
     ]
    }
   ],
   "source": [
    "clustalw_exe = r\"/home/user1/Applications/clustalw/clustalw2\"\n",
    "clustalw_cline = ClustalwCommandline(clustalw_exe, infile=\"secuencias_anotadas.fasta\")\n",
    "assert os.path.isfile(clustalw_exe), \"Clustal W executable missing\"\n",
    "stdout, stderr = clustalw_cline()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "ename": "ApplicationError",
     "evalue": "Non-zero return code 126 from '/home/user1/Applications/clustalw/clustalw2 -infile=secuencias_comp.fas', message '/bin/sh: 1: /home/user1/Applications/clustalw/clustalw2: Permission denied'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m\u001b[0m",
      "\u001b[0;31mApplicationError\u001b[0mTraceback (most recent call last)",
      "\u001b[0;32m<ipython-input-36-b84c0ec8a6ae>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      2\u001b[0m \u001b[0mclustalw_cline\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mClustalwCommandline\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mclustalw_exe\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0minfile\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m\"secuencias_comp.fas\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      3\u001b[0m \u001b[0;32massert\u001b[0m \u001b[0mos\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpath\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0misfile\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mclustalw_exe\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m\"Clustal W executable missing\"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 4\u001b[0;31m \u001b[0mstdout\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstderr\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mclustalw_cline\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m/home/jupyter/miniconda3/lib/python3.6/site-packages/Bio/Application/__init__.py\u001b[0m in \u001b[0;36m__call__\u001b[0;34m(self, stdin, stdout, stderr, cwd, env)\u001b[0m\n\u001b[1;32m    521\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mreturn_code\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    522\u001b[0m             raise ApplicationError(return_code, str(self),\n\u001b[0;32m--> 523\u001b[0;31m                                    stdout_str, stderr_str)\n\u001b[0m\u001b[1;32m    524\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mstdout_str\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstderr_str\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    525\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mApplicationError\u001b[0m: Non-zero return code 126 from '/home/user1/Applications/clustalw/clustalw2 -infile=secuencias_comp.fas', message '/bin/sh: 1: /home/user1/Applications/clustalw/clustalw2: Permission denied'"
     ]
    }
   ],
   "source": [
    "clustalw_exe = r\"/home/user1/Applications/clustalw/clustalw2\"\n",
    "clustalw_cline = ClustalwCommandline(clustalw_exe, infile=\"secuencias_comp.fas\")\n",
    "assert os.path.isfile(clustalw_exe), \"Clustal W executable missing\"\n",
    "stdout, stderr = clustalw_cline()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;34mimg\u001b[0m/  secuencias_comp.fas\r\n"
     ]
    }
   ],
   "source": [
    "ls "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "os.path.isfile('/home/user1/Applications/clustalw/clustalw2')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'~/Applications/clustalw/clustalw2'"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clustalw_exe\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "alignments = AlignIO.parse(\"../fasta/secuencias_alinear.aln\", \"clustal\")\n",
    "for alignment in alignments:\n",
    "    print(alignment)\n",
    "    print(\"\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "# para obtener ayuda en clustalw"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "\r\n",
      "\r\n",
      " CLUSTAL 2.1 Multiple Sequence Alignments\r\n",
      "\r\n",
      "\r\n",
      "                DATA (sequences)\r\n",
      "\r\n",
      "-INFILE=file.ext                             :input sequences.\r\n",
      "-PROFILE1=file.ext  and  -PROFILE2=file.ext  :profiles (old alignment).\r\n",
      "\r\n",
      "\r\n",
      "                VERBS (do things)\r\n",
      "\r\n",
      "-OPTIONS            :list the command line parameters\r\n",
      "-HELP  or -CHECK    :outline the command line params.\r\n",
      "-FULLHELP           :output full help content.\r\n",
      "-ALIGN              :do full multiple alignment.\r\n",
      "-TREE               :calculate NJ tree.\r\n",
      "-PIM                :output percent identity matrix (while calculating the tree)\r\n",
      "-BOOTSTRAP(=n)      :bootstrap a NJ tree (n= number of bootstraps; def. = 1000).\r\n",
      "-CONVERT            :output the input sequences in a different file format.\r\n",
      "\r\n",
      "\r\n",
      "                PARAMETERS (set things)\r\n",
      "\r\n",
      "***General settings:****\r\n",
      "-INTERACTIVE :read command line, then enter normal interactive menus\r\n",
      "-QUICKTREE   :use FAST algorithm for the alignment guide tree\r\n",
      "-TYPE=       :PROTEIN or DNA sequences\r\n",
      "-NEGATIVE    :protein alignment with negative values in matrix\r\n",
      "-OUTFILE=    :sequence alignment file name\r\n",
      "-OUTPUT=     :CLUSTAL(default), GCG, GDE, PHYLIP, PIR, NEXUS and FASTA\r\n",
      "-OUTORDER=   :INPUT or ALIGNED\r\n",
      "-CASE        :LOWER or UPPER (for GDE output only)\r\n",
      "-SEQNOS=     :OFF or ON (for Clustal output only)\r\n",
      "-SEQNO_RANGE=:OFF or ON (NEW: for all output formats)\r\n",
      "-RANGE=m,n   :sequence range to write starting m to m+n\r\n",
      "-MAXSEQLEN=n :maximum allowed input sequence length\r\n",
      "-QUIET       :Reduce console output to minimum\r\n",
      "-STATS=      :Log some alignents statistics to file\r\n",
      "\r\n",
      "***Fast Pairwise Alignments:***\r\n",
      "-KTUPLE=n    :word size\r\n",
      "-TOPDIAGS=n  :number of best diags.\r\n",
      "-WINDOW=n    :window around best diags.\r\n",
      "-PAIRGAP=n   :gap penalty\r\n",
      "-SCORE       :PERCENT or ABSOLUTE\r\n",
      "\r\n",
      "\r\n",
      "***Slow Pairwise Alignments:***\r\n",
      "-PWMATRIX=    :Protein weight matrix=BLOSUM, PAM, GONNET, ID or filename\r\n",
      "-PWDNAMATRIX= :DNA weight matrix=IUB, CLUSTALW or filename\r\n",
      "-PWGAPOPEN=f  :gap opening penalty        \r\n",
      "-PWGAPEXT=f   :gap opening penalty\r\n",
      "\r\n",
      "\r\n",
      "***Multiple Alignments:***\r\n",
      "-NEWTREE=      :file for new guide tree\r\n",
      "-USETREE=      :file for old guide tree\r\n",
      "-MATRIX=       :Protein weight matrix=BLOSUM, PAM, GONNET, ID or filename\r\n",
      "-DNAMATRIX=    :DNA weight matrix=IUB, CLUSTALW or filename\r\n",
      "-GAPOPEN=f     :gap opening penalty        \r\n",
      "-GAPEXT=f      :gap extension penalty\r\n",
      "-ENDGAPS       :no end gap separation pen. \r\n",
      "-GAPDIST=n     :gap separation pen. range\r\n",
      "-NOPGAP        :residue-specific gaps off  \r\n",
      "-NOHGAP        :hydrophilic gaps off\r\n",
      "-HGAPRESIDUES= :list hydrophilic res.    \r\n",
      "-MAXDIV=n      :% ident. for delay\r\n",
      "-TYPE=         :PROTEIN or DNA\r\n",
      "-TRANSWEIGHT=f :transitions weighting\r\n",
      "-ITERATION=    :NONE or TREE or ALIGNMENT\r\n",
      "-NUMITER=n     :maximum number of iterations to perform\r\n",
      "-NOWEIGHTS     :disable sequence weighting\r\n",
      "\r\n",
      "\r\n",
      "***Profile Alignments:***\r\n",
      "-PROFILE      :Merge two alignments by profile alignment\r\n",
      "-NEWTREE1=    :file for new guide tree for profile1\r\n",
      "-NEWTREE2=    :file for new guide tree for profile2\r\n",
      "-USETREE1=    :file for old guide tree for profile1\r\n",
      "-USETREE2=    :file for old guide tree for profile2\r\n",
      "\r\n",
      "\r\n",
      "***Sequence to Profile Alignments:***\r\n",
      "-SEQUENCES   :Sequentially add profile2 sequences to profile1 alignment\r\n",
      "-NEWTREE=    :file for new guide tree\r\n",
      "-USETREE=    :file for old guide tree\r\n",
      "\r\n",
      "\r\n",
      "***Structure Alignments:***\r\n",
      "-NOSECSTR1     :do not use secondary structure-gap penalty mask for profile 1 \r\n",
      "-NOSECSTR2     :do not use secondary structure-gap penalty mask for profile 2\r\n",
      "-SECSTROUT=STRUCTURE or MASK or BOTH or NONE   :output in alignment file\r\n",
      "-HELIXGAP=n    :gap penalty for helix core residues \r\n",
      "-STRANDGAP=n   :gap penalty for strand core residues\r\n",
      "-LOOPGAP=n     :gap penalty for loop regions\r\n",
      "-TERMINALGAP=n :gap penalty for structure termini\r\n",
      "-HELIXENDIN=n  :number of residues inside helix to be treated as terminal\r\n",
      "-HELIXENDOUT=n :number of residues outside helix to be treated as terminal\r\n",
      "-STRANDENDIN=n :number of residues inside strand to be treated as terminal\r\n",
      "-STRANDENDOUT=n:number of residues outside strand to be treated as terminal \r\n",
      "\r\n",
      "\r\n",
      "***Trees:***\r\n",
      "-OUTPUTTREE=nj OR phylip OR dist OR nexus\r\n",
      "-SEED=n        :seed number for bootstraps.\r\n",
      "-KIMURA        :use Kimura's correction.   \r\n",
      "-TOSSGAPS      :ignore positions with gaps.\r\n",
      "-BOOTLABELS=node OR branch :position of bootstrap values in tree display\r\n",
      "-CLUSTERING=   :NJ or UPGMA\r\n"
     ]
    }
   ],
   "source": [
    "!/Applications/clustalw/clustalw2 -help"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Help on package Bio.AlignIO in Bio:\n",
      "\n",
      "NAME\n",
      "    Bio.AlignIO - Multiple sequence alignment input/output as alignment objects.\n",
      "\n",
      "DESCRIPTION\n",
      "    The Bio.AlignIO interface is deliberately very similar to Bio.SeqIO, and in\n",
      "    fact the two are connected internally.  Both modules use the same set of file\n",
      "    format names (lower case strings).  From the user's perspective, you can read\n",
      "    in a PHYLIP file containing one or more alignments using Bio.AlignIO, or you\n",
      "    can read in the sequences within these alignmenta using Bio.SeqIO.\n",
      "    \n",
      "    Bio.AlignIO is also documented at http://biopython.org/wiki/AlignIO and by\n",
      "    a whole chapter in our tutorial:\n",
      "    \n",
      "    * `HTML Tutorial`_\n",
      "    * `PDF Tutorial`_\n",
      "    \n",
      "    .. _`HTML Tutorial`: http://biopython.org/DIST/docs/tutorial/Tutorial.html\n",
      "    .. _`PDF Tutorial`: http://biopython.org/DIST/docs/tutorial/Tutorial.pdf\n",
      "    \n",
      "    Input\n",
      "    -----\n",
      "    For the typical special case when your file or handle contains one and only\n",
      "    one alignment, use the function Bio.AlignIO.read().  This takes an input file\n",
      "    handle (or in recent versions of Biopython a filename as a string), format\n",
      "    string and optional number of sequences per alignment.  It will return a single\n",
      "    MultipleSeqAlignment object (or raise an exception if there isn't just one\n",
      "    alignment):\n",
      "    \n",
      "    >>> from Bio import AlignIO\n",
      "    >>> align = AlignIO.read(\"Phylip/interlaced.phy\", \"phylip\")\n",
      "    >>> print(align)\n",
      "    SingleLetterAlphabet() alignment with 3 rows and 384 columns\n",
      "    -----MKVILLFVLAVFTVFVSS---------------RGIPPE...I-- CYS1_DICDI\n",
      "    MAHARVLLLALAVLATAAVAVASSSSFADSNPIRPVTDRAASTL...VAA ALEU_HORVU\n",
      "    ------MWATLPLLCAGAWLLGV--------PVCGAAELSVNSL...PLV CATH_HUMAN\n",
      "    \n",
      "    For the general case, when the handle could contain any number of alignments,\n",
      "    use the function Bio.AlignIO.parse(...) which takes the same arguments, but\n",
      "    returns an iterator giving MultipleSeqAlignment objects (typically used in a\n",
      "    for loop). If you want random access to the alignments by number, turn this\n",
      "    into a list:\n",
      "    \n",
      "    >>> from Bio import AlignIO\n",
      "    >>> alignments = list(AlignIO.parse(\"Emboss/needle.txt\", \"emboss\"))\n",
      "    >>> print(alignments[2])\n",
      "    SingleLetterAlphabet() alignment with 2 rows and 120 columns\n",
      "    -KILIVDDQYGIRILLNEVFNKEGYQTFQAANGLQALDIVTKER...--- ref_rec\n",
      "    LHIVVVDDDPGTCVYIESVFAELGHTCKSFVRPEAAEEYILTHP...HKE gi|94967506|receiver\n",
      "    \n",
      "    Most alignment file formats can be concatenated so as to hold as many\n",
      "    different multiple sequence alignments as possible.  One common example\n",
      "    is the output of the tool seqboot in the PHLYIP suite.  Sometimes there\n",
      "    can be a file header and footer, as seen in the EMBOSS alignment output.\n",
      "    \n",
      "    Output\n",
      "    ------\n",
      "    Use the function Bio.AlignIO.write(...), which takes a complete set of\n",
      "    Alignment objects (either as a list, or an iterator), an output file handle\n",
      "    (or filename in recent versions of Biopython) and of course the file format::\n",
      "    \n",
      "      from Bio import AlignIO\n",
      "      alignments = ...\n",
      "      count = SeqIO.write(alignments, \"example.faa\", \"fasta\")\n",
      "    \n",
      "    If using a handle make sure to close it to flush the data to the disk::\n",
      "    \n",
      "      from Bio import AlignIO\n",
      "      alignments = ...\n",
      "      with open(\"example.faa\", \"w\") as handle:\n",
      "        count = SeqIO.write(alignments, handle, \"fasta\")\n",
      "    \n",
      "    In general, you are expected to call this function once (with all your\n",
      "    alignments) and then close the file handle.  However, for file formats\n",
      "    like PHYLIP where multiple alignments are stored sequentially (with no file\n",
      "    header and footer), then multiple calls to the write function should work as\n",
      "    expected when using handles.\n",
      "    \n",
      "    If you are using a filename, the repeated calls to the write functions will\n",
      "    overwrite the existing file each time.\n",
      "    \n",
      "    Conversion\n",
      "    ----------\n",
      "    The Bio.AlignIO.convert(...) function allows an easy interface for simple\n",
      "    alignment file format conversions. Additionally, it may use file format\n",
      "    specific optimisations so this should be the fastest way too.\n",
      "    \n",
      "    In general however, you can combine the Bio.AlignIO.parse(...) function with\n",
      "    the Bio.AlignIO.write(...) function for sequence file conversion. Using\n",
      "    generator expressions provides a memory efficient way to perform filtering or\n",
      "    other extra operations as part of the process.\n",
      "    \n",
      "    File Formats\n",
      "    ------------\n",
      "    When specifying the file format, use lowercase strings.  The same format\n",
      "    names are also used in Bio.SeqIO and include the following:\n",
      "    \n",
      "      - clustal -   Output from Clustal W or X, see also the module Bio.Clustalw\n",
      "        which can be used to run the command line tool from Biopython.\n",
      "      - emboss    - EMBOSS tools' \"pairs\" and \"simple\" alignment formats.\n",
      "      - fasta     - The generic sequence file format where each record starts with\n",
      "        an identifer line starting with a \">\" character, followed by\n",
      "        lines of sequence.\n",
      "      - fasta-m10 - For the pairswise alignments output by Bill Pearson's FASTA\n",
      "        tools when used with the -m 10 command line option for machine\n",
      "        readable output.\n",
      "      - ig        - The IntelliGenetics file format, apparently the same as the\n",
      "        MASE alignment format.\n",
      "      - nexus     - Output from NEXUS, see also the module Bio.Nexus which can also\n",
      "        read any phylogenetic trees in these files.\n",
      "      - phylip    - Interlaced PHYLIP, as used by the PHLIP tools.\n",
      "      - phylip-sequential - Sequential PHYLIP.\n",
      "      - phylip-relaxed - PHYLIP like format allowing longer names.\n",
      "      - stockholm - A richly annotated alignment file format used by PFAM.\n",
      "    \n",
      "    Note that while Bio.AlignIO can read all the above file formats, it cannot\n",
      "    write to all of them.\n",
      "    \n",
      "    You can also use any file format supported by Bio.SeqIO, such as \"fasta\" or\n",
      "    \"ig\" (which are listed above), PROVIDED the sequences in your file are all the\n",
      "    same length.\n",
      "\n",
      "PACKAGE CONTENTS\n",
      "    ClustalIO\n",
      "    EmbossIO\n",
      "    FastaIO\n",
      "    Interfaces\n",
      "    MafIO\n",
      "    NexusIO\n",
      "    PhylipIO\n",
      "    StockholmIO\n",
      "\n",
      "FUNCTIONS\n",
      "    convert(in_file, in_format, out_file, out_format, alphabet=None)\n",
      "        Convert between two alignment files, returns number of alignments.\n",
      "        \n",
      "            - in_file - an input handle or filename\n",
      "            - in_format - input file format, lower case string\n",
      "            - output - an output handle or filename\n",
      "            - out_file - output file format, lower case string\n",
      "            - alphabet - optional alphabet to assume\n",
      "        \n",
      "        **NOTE** - If you provide an output filename, it will be opened which will\n",
      "        overwrite any existing file without warning. This may happen if even the\n",
      "        conversion is aborted (e.g. an invalid out_format name is given).\n",
      "    \n",
      "    parse(handle, format, seq_count=None, alphabet=None)\n",
      "        Iterate over an alignment file as MultipleSeqAlignment objects.\n",
      "        \n",
      "        Arguments:\n",
      "          - handle    - handle to the file, or the filename as a string\n",
      "            (note older versions of Biopython only took a handle).\n",
      "          - format    - string describing the file format.\n",
      "          - alphabet  - optional Alphabet object, useful when the sequence type\n",
      "            cannot be automatically inferred from the file itself\n",
      "            (e.g. fasta, phylip, clustal)\n",
      "          - seq_count - Optional integer, number of sequences expected in each\n",
      "            alignment.  Recommended for fasta format files.\n",
      "        \n",
      "        If you have the file name in a string 'filename', use:\n",
      "        \n",
      "        >>> from Bio import AlignIO\n",
      "        >>> filename = \"Emboss/needle.txt\"\n",
      "        >>> format = \"emboss\"\n",
      "        >>> for alignment in AlignIO.parse(filename, format):\n",
      "        ...     print(\"Alignment of length %i\" % alignment.get_alignment_length())\n",
      "        Alignment of length 124\n",
      "        Alignment of length 119\n",
      "        Alignment of length 120\n",
      "        Alignment of length 118\n",
      "        Alignment of length 125\n",
      "        \n",
      "        If you have a string 'data' containing the file contents, use::\n",
      "        \n",
      "          from Bio import AlignIO\n",
      "          from StringIO import StringIO\n",
      "          my_iterator = AlignIO.parse(StringIO(data), format)\n",
      "        \n",
      "        Use the Bio.AlignIO.read() function when you expect a single record only.\n",
      "    \n",
      "    read(handle, format, seq_count=None, alphabet=None)\n",
      "        Turns an alignment file into a single MultipleSeqAlignment object.\n",
      "        \n",
      "        Arguments:\n",
      "          - handle    - handle to the file, or the filename as a string\n",
      "            (note older versions of Biopython only took a handle).\n",
      "          - format    - string describing the file format.\n",
      "          - alphabet  - optional Alphabet object, useful when the sequence type\n",
      "            cannot be automatically inferred from the file itself\n",
      "            (e.g. fasta, phylip, clustal)\n",
      "          - seq_count - Optional integer, number of sequences expected in each\n",
      "            alignment.  Recommended for fasta format files.\n",
      "        \n",
      "        If the handle contains no alignments, or more than one alignment,\n",
      "        an exception is raised.  For example, using a PFAM/Stockholm file\n",
      "        containing one alignment:\n",
      "        \n",
      "        >>> from Bio import AlignIO\n",
      "        >>> filename = \"Clustalw/protein.aln\"\n",
      "        >>> format = \"clustal\"\n",
      "        >>> alignment = AlignIO.read(filename, format)\n",
      "        >>> print(\"Alignment of length %i\" % alignment.get_alignment_length())\n",
      "        Alignment of length 411\n",
      "        \n",
      "        If however you want the first alignment from a file containing\n",
      "        multiple alignments this function would raise an exception.\n",
      "        \n",
      "        >>> from Bio import AlignIO\n",
      "        >>> filename = \"Emboss/needle.txt\"\n",
      "        >>> format = \"emboss\"\n",
      "        >>> alignment = AlignIO.read(filename, format)\n",
      "        Traceback (most recent call last):\n",
      "            ...\n",
      "        ValueError: More than one record found in handle\n",
      "        \n",
      "        Instead use:\n",
      "        \n",
      "        >>> from Bio import AlignIO\n",
      "        >>> filename = \"Emboss/needle.txt\"\n",
      "        >>> format = \"emboss\"\n",
      "        >>> alignment = next(AlignIO.parse(filename, format))\n",
      "        >>> print(\"First alignment has length %i\" % alignment.get_alignment_length())\n",
      "        First alignment has length 124\n",
      "        \n",
      "        You must use the Bio.AlignIO.parse() function if you want to read multiple\n",
      "        records from the handle.\n",
      "    \n",
      "    write(alignments, handle, format)\n",
      "        Write complete set of alignments to a file.\n",
      "        \n",
      "        Arguments:\n",
      "          - alignments - A list (or iterator) of MultipleSeqAlignment objects,\n",
      "            or a single alignment object.\n",
      "          - handle    - File handle object to write to, or filename as string\n",
      "            (note older versions of Biopython only took a handle).\n",
      "          - format    - lower case string describing the file format to write.\n",
      "        \n",
      "        You should close the handle after calling this function.\n",
      "        \n",
      "        Returns the number of alignments written (as an integer).\n",
      "\n",
      "DATA\n",
      "    print_function = _Feature((2, 6, 0, 'alpha', 2), (3, 0, 0, 'alpha', 0)...\n",
      "\n",
      "FILE\n",
      "    /Users/migueldelrio/anaconda3/lib/python3.6/site-packages/Bio/AlignIO/__init__.py\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "help(AlignIO)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Nav1.2F.fa                PET1.8F.fasta             PETsn1.8R.fasta\r\n",
      "Nav1.2F.fasta             PET1.8R.fasta             PHBb1.2F.fasta\r\n",
      "Nav1.2R.fasta             PET2.2F.fasta             PHBb1.2R.fasta\r\n",
      "Nav1.8F.fasta             PET2.2R.fasta             secuencias.fasta\r\n",
      "Nav1.8R.fasta             PET2.8F.fasta             secuencias_alinear.aln\r\n",
      "Nav2.2F.fasta             PET2.8R.fasta             secuencias_alinear.dnd\r\n",
      "Nav2.2R.fasta             PETsn1.2F.fasta           secuencias_alinear.fasta\r\n",
      "PET1.2F.fasta             PETsn1.2R.fasta\r\n",
      "PET1.2R.fasta             PETsn1.8F.fasta\r\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SingleLetterAlphabet() alignment with 20 rows and 378 columns\n",
      "---------------CCGCAGGTTCCCCTAGGGTTACCTTGTTA...--- PET2.2RC\n",
      "-----GGTGATCCAACCGCAGGTTCCCCTAGGGTTACCTTGTTA...--- PETsn1.8RC\n",
      "---------------CCGCAGGTTCCCCTAGGGTTACCTTGTTA...--- Nav2.2RC\n",
      "----------------CGCAGGTTCCCCTAGGGTTACCTTGTTA...--- PETsn1.2RC\n",
      "-----------------GCAGGTTCCCCTAGGGTTACCTTGTTA...--- PET2.8RC\n",
      "--------------------------------------------...T-- Nav2.2F\n",
      "-------------------------------GGATTTTGCCGGC...TA- PETsn1.8F\n",
      "--------------------------------------------...T-- PET2.8F\n",
      "--------------------------------------------...TAA PET2.2F\n",
      "--------------------------------------------...--- PETsn1.2F\n",
      "--------------------------------------------...--- PET1.2F\n",
      "--------------------------------------------...--- PET1.8F\n",
      "-------------------AGGTTCCCCTACGGCTACCTTGTTA...--- PET1.2RC\n",
      "AGGAGGTGGAATCAACCGCAGGTTCCCCTACGGCTACCTTGTTA...--- PET1.8RC\n",
      "--------------------------------------------...--- Nav1.2F\n",
      "--------------------------------------------...--- Nav1.8F\n",
      "-----------------GCACCTTCCGATACGGCTACCTTGTTA...--- Nav1.2RC\n",
      "-----------------GCACCTTCCGATACGGTTACCTTGTTA...--- Nav1.8RC\n",
      "-------------------------------------------T...--- PHBb1.2F\n",
      "-----------------GCAGGTTCCCCTACGGCTACCTTGTTA...--- PHBb1.2RC\n",
      "\n"
     ]
    }
   ],
   "source": [
    "alignments = AlignIO.parse(\"secuencias_alinear.aln\", \"clustal\")\n",
    "for alignment in alignments:\n",
    "    print(alignment)\n",
    "    print(\"\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import Phylo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                                   ______________ Nav1.2F\n",
      "                                             _____|\n",
      "                                            |     |_ Nav1.8F\n",
      "                                ____________|\n",
      "                               |            |     _______ Nav1.2RC\n",
      "                               |            |____|\n",
      "                      _________|                 |____ Nav1.8RC\n",
      "                     |         |\n",
      "                     |         |         _____ PHBb1.2F\n",
      "                     |         |________|\n",
      "           __________|                  |_______ PHBb1.2RC\n",
      "          |          |\n",
      "          |          |     ___ PET1.2F\n",
      "          |          | ___|\n",
      "          |          ||   |____ PET1.8F\n",
      "          |          ||\n",
      "          |           |    ___ PET1.2RC\n",
      "          |           |___|\n",
      "  ________|               |____ PET1.8RC\n",
      " |        |\n",
      " |        |           , Nav2.2F\n",
      " |        |          ,|\n",
      " |        |         _||__ PETsn1.8F\n",
      " |        |        | |\n",
      " |        |       ,| | PET2.8F\n",
      " |        |       ||\n",
      " |        |_______|| PET2.2F\n",
      " |                |\n",
      " |                | PETsn1.2F\n",
      " |\n",
      " | PET2.8RC\n",
      "_|\n",
      " |, Nav2.2RC\n",
      " ||\n",
      " |, PET2.2RC\n",
      " ||\n",
      " ||_____ PETsn1.8RC\n",
      " |\n",
      " | PETsn1.2RC\n",
      "\n"
     ]
    }
   ],
   "source": [
    "tree = Phylo.read(\"secuencias_alinear.dnd\", \"newick\")\n",
    "Phylo.draw_ascii(tree, file=None, column_width=80)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAYwAAAEKCAYAAAAB0GKPAAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAAIABJREFUeJzt3XtcVVX6+PHPkyRq4d3CG6FhCgIe0dFw8lYjo+RMjSBO\nmmLqGJnlWJLlaN/0q1noqGM1P6cw/aJNXqcxjdFJ8pKWlRe8ZKbOSImSkWNIoQm6fn+cwxkuBzjA\nOYfb8369eHnO2mvvvfbOWO611/MsMcaglFJKleWmqm6AUkqpmkE7DKWUUk7RDkMppZRTtMNQSinl\nFO0wlFJKOUU7DKWUUk7RDkMppZRTtMNQSinlFO0wlFJKOcWrqhvgSi1btjT+/v5V3QyllKoxDhw4\n8J0xppUzdWtVh+Hv78/+/furuhlKKVVjiMhXztbVISmllFJO0Q5DKaWUU7TDUEqpAvz9/YmKirJ/\n37BhA2PHjq3QsU6cOEF4eDje3t4sXLiwxHqjRo2ic+fOBAcHM27cOHJzcwFYuXIlrVq1wmKxYLFY\nGDNmTIXa4SraYSilVBEHDhzg+PHjlT5O8+bNWbp0KdOmTSu13qhRozhx4gRHjx7lypUrJCYm2reN\nGDGC1NRUUlNTSUpKqnSbKkM7DKWUKuLpp59m3rx5xco//fRTwsPD6d69O3369OHLL78E4O677+bz\nzz+31xswYAD79+/ntttu42c/+xk333xzqeeLjIxERBARevXqRXp6umsvyEW0w1BKqSJiYmI4ePAg\np0+fLlTepUsXPvzwQw4dOsScOXOYMWMGYH0KWLduHQAZGRlkZGTQs2fPcp83NzeXVatWMXjwYHvZ\n2rVr7UNSK1asqMRVVV6tmlarlHKtAQMGVHUT3G7nzp3FyurVq0d8fDzz589nyJAh9vKsrCxiY2M5\ndeoUImJ/1xATE0NERASzZ89m3bp1REdHV6gtkyZNol+/fvTt29deNmLECF599dUKHc/V9AlDKeXQ\ngAEDSE1NrepmVJnRo0eze/duzp49ay+bNWsWAwcO5NixY2zevJmrV68C0LZtW1q0aMGRI0dYu3Yt\nI0aMKPf5Zs+eTWZmJosWLXLZNbiaPmEopUpksVgc/gu8Lrj55puZOnUqL730Evfeey9gfcJo27Yt\nYJ3BVNCIESNISEggKyuL0NDQcp0rMTGRbdu2kZKSwk03Vd9/x1fflimlVBUbP348eXl59u/PPPMM\nzz33HN27dy9UDhAdHc2aNWuIiYmxl33zzTe0a9eORYsWMXfuXNq1a8fly5cB64vu8+fPAxAXF8eF\nCxcIDw/HYrEwZ84cD1xd+Ykxpqrb4DI9e/Y0mhpEKdfIf39RV58w6goROWCMceoNvQ5JKVXDeOpF\ndGpqKhaLxSPnUjWDDkkpVYPU9RfRqmrpE4ZSNYynXkRX9EnG39+fHj16sHHjRsCaWmPLli3FXhI7\n48SJEzzyyCMcPHiQefPmlRgxnZKSQnx8PDdu3ODWW29l5cqVBAQEsHLlSuLj4+0vqkNDQ6s8Wrom\n0ycMpZTLeTq1xmOPPcZbb71FamoqI0eOZO7cufZt1Sm1Rk2nHYZSyuU8nVpDROyzj7KysmjTpo0L\nr0bl0yEppVSJUlNTyxyacjQ8FhMTw5///OcSU2t4eXmxfft2ZsyYwcaNG+2pNWbPnl2h1BqJiYlE\nRkbSsGFDGjduzL59++zb1q5dy549ewCYMmUKjzzyiNPHVYXpE4ZSyqGdO3dWeJZUwdQaBWVlZTF8\n+HCCg4OZOnWq/akiJiaGDRs2AFQotcbixYtJTk4mPT2dRx55hKeeesq+reCQlHYWlaNPGEqpElXm\n5fro0aOZP38+wcHB9rL81BrvvPMOaWlp9qeXoqk1li1b5vR5MjMzOXz4ML179wasHUTB5H3KdTz+\nhCEi7UVkh4gcF5HPRWSKgzoDRCRLRFJtP897up1KqcrJT62xePFie5k7Ums0a9aMrKwsTp48CcD7\n779PYGBg5S9AFVMVQ1J5wNPGmCDgbuBxEQlyUO9DY4zF9lM94+SVUqXyRGoNLy8v3njjDaKioujW\nrRurVq1iwYIFnrnAOqbKU4OIyCbgVWPM+wXKBgDTjDFDy3MsTQ2iajtN16FcrcakBhERf6A78ImD\nzX1E5AhwDmvn8bmDOkqVqTat6aDpOlRVqrJZUiJyK7AR+L0x5nKRzQcBP2NMKPAK8PdSjjNRRPaL\nyP7MzEz3NVjVSJpKo+by9/cnKirK/n3Dhg2MHTu2Qsc6ceIE4eHheHt7s3DhwhLrjRo1is6dOxMc\nHMy4cePsCyStXLmSVq1aYbFY6NKlS6H3MgBJSUkEBwcTEhJC9+7dSz1HTVYlTxgicjPWzuItY8zf\nim4v2IEYY5JF5M8i0tIY852Duq8Dr4N1SMqNzVY1VG1a06E2PS05Iz9iPCjI0WtO5+VHjP/97yX+\n2xOwdhirV68GYOTIkSQmJvLYY48B/1357uLFi3Tu3Jno6Gjat2/PP/7xD5YsWcI///lP2rRpw08/\n/VRrI8qrYpaUAMuBL4wxDpeWEhFfWz1EpBfWdl70XCuVUtWBpyPGIyMjERFEhF69epGenl6sTosW\nLQgICCAjIwOA+fPns3DhQnt0ube3N7/73e8qfM3VWVUMSf0cGA3cW2DabKSIxIlInK1ONHBMRA4D\nS4Hfmqp+O6+U8riYmBgOHjxYYsT4oUOHmDNnDjNmzACwR4wDFYoYz5ebm8uqVascxnN8/fXXXL16\n1T7199ixY/To0aPc56iJPD4kZYzZA0gZdV4Fqseq50rVIVU55OVo2LBgxPiQIUPs5VlZWcTGxnLq\n1ClExP6uISYmhoiICGbPnl2hiPF8kyZNol+/fvTt29detnbtWnbv3s2JEyd49dVXadCgQYWOXZNp\nahClFFB9JwiMHj2a3bt3c/bsWXtZfsT4sWPH2Lx5M1evXgWKR4yPGDGi3OebPXs2mZmZLFpUeMR8\nxIgRHDlyhI8++ohnn32Wb775BoCuXbty4MCBSlxhzaGpQZRSdtVxgkB+xPhLL73EvffeC7gnYhys\nSQy3bdtGSkoKN93k+N/TPXv2ZPTo0fzpT39i/vz5PPfcc8THx/Pee+/h6+vLtWvXSEpKYsKECeW/\n2GpOnzCUUtWeJyLGAeLi4rhw4QLh4eFYLBbmzHGcZGL69OmsWLGC7OxsIiMjmTx5Mr/4xS/o2rUr\nYWFh9mPXNlUe6e1KGumtiqptkdHuvJ7adq+Uc2pMpLdSqvycWaOiosfVKHJVGh2SUqoGqcwaFUpV\nlj5hKFXDuGvIyJ1Tav39/enRowcbN24ErGk+tmzZUuyFtTNOnDjBI488wsGDB5k3b16J632npKQQ\nHx/PjRs3uPXWW1m5ciUBAQGsXLmS+Ph42rZty9WrV3n00UeZOnWqfb+kpCQSEhIQEby8vBg1alSZ\na4rXFfqEoZTyiPw0H5WVn+ajrF/ijz32GG+99RapqamMHDmSuXPn2rflr8K3d+9e5s2bZ5+yWzDN\nx9GjR9m3bx9NmjSpdJtrC+0wlFIe4ek0HyJin62UlZVlT91RUF1O81EROiSlKqW6J8PTF7nl46oX\n6o6GzWJiYvjzn/9cYpoPLy8vtm/fzowZM9i4caM9zcfs2bMrlOYjMTGRyMhIGjZsSOPGjdm3b1+x\nOnU5zUdF6BOGqrDqGhmsKsbdL9QLpvkoKCsri+HDhxMcHMzUqVPtTxUxMTFs2LABoEJpPhYvXkxy\ncjLp6ek88sgjPPXUU/Zta9euJTQ0lICAACZNmlQn03xUhD5hqEqpjpHBBVX3J6Dqxt3/LUePHs38\n+fMJDg62l+Wn+XjnnXdIS0uz/zcrmuZj2bJlTp8nMzOTw4cP07t3b8D6zqJgIsH8VOX79+8nIiKC\nX//61/j6+trTfORHlKvC9AlDKeUx+Wk+Ci5A5I40H82aNSMrK4uTJ08C8P777xMYGFisXsE0H4A9\nzUd+nqhr166RmJhYrmuszbTDUEp5lCfSfHh5efHGG28QFRVFt27dWLVqFQsWLHDYnrqa5qMiNDWI\nqrCakEqiJrRRqapUntQg+oShlFLKKdphKFWH+Pv7ExISQmhoKBEREfaxen9/f7777jt7vZ07dzJ0\n6FDA+l6hVatWWCwWunbtSnR0NDk5OQCMHTvWPpOpoIsXLzJw4EBuvfVWJk+eXGJ74uPj6dKlC6Gh\nofzmN7/h+++/t5+/SZMmWCwWLBYLv/jFL1x2D1TFVUmHISJpInLUtjxrsTEksVoqIqdF5IiIhFVF\nO5WqjXbs2MGRI0fo2bMnL774olP75EdGf/7559SvX5+1a9eWWr9Bgwb87//+LwsXLiy13qBBgzh2\n7BhHjhzhrrvuKjTltm/fvqSmppKamsr27dudaqdyr6p8whhojLGUMHY2BOhk+5kI/D+PtkypOqBf\nv37FgujKkpeXx48//kizZs3sZdu3b6dnz57cddddbNmyBYBbbrmFe+65p8z4hoiICLy8rLP77777\nbtLT08t5FcqTqmscxgNAkrG+kd8nIk1FpLUxJqOqG6ZUbbFlyxZCQkLs3wcOHEi9evUA+OGHH+jS\npYt929q1a9mzZw8ZGRncdddd/OpXv7JvS0tL49NPP+Vf//oXAwcO5PTp0xUKhHvzzTcLLan64Ycf\n2gMJhw8fzh/+8IdyH1O5VlV1GAbYLiLXgb8YY14vsr0tcLbA93RbmXYYqlbwVECho9lh+R1DaGho\noYR8O3bsoGXLlvb9Cg4n5Qe6GWN4/PHHWbBgAc8++yxgjci+6aab6NSpEx07duTEiRPljhifN2+e\nPTNsvr59+9qfWFT1UFVDUvcYYyxYh54eF5F+FT2QiEwUkf0isj8zM9N1LVTKTao6pcqOHTtITU0l\nKSmJpk2blmtfEeFXv/oVu3fvLlRWtE55rFy5ki1btvDWW2+Ve1/lWVXyhGGMOWf781sReQfoBewu\nUOUc0L7A93a2MkfHeh14HaxxGG5psFIuVt1TqpRmz5493Hnnnfbv69evJzY2ljNnzvDvf/+bzp07\nO32srVu3kpCQwK5du2jUqJE7mqtcyOMdhojcAtxkjMm2fY4Aiq60/i4wWUTWAL2BLH1/oVTVyX+H\ncePGDdq1a1cohYefnx+9evXi8uXLLFu2zP7+wt/fn8uXL3Pt2jX+/ve/889//pOgoCAmTJhAXFwc\nPXv2ZPLkyfz0008MGjQIsL74Lk/OKOVZHo/0FpGOwDu2r17AX40x80QkDsAYs0ysz6WvAoOBHOAR\nY0yZIdwa6e1ZNSGKujq2sTq2SdVd5Yn09vgThjHm30A3B+XLCnw2wOOebJeqvVy1xoOr6BodqqbS\nSG9Vq7l7jQel6hRjTK356dGjh1Ge079/f9O/f/+qbkaNUx3v2x133GGCg4NNSEiIGTRokMnIyLCX\nZ2Zm2uvt2LHD3H///cYYY1asWGFatmxpunXrZoKCgkxUVJT58ccfjTHGxMbGmvXr1xc7z3fffWcG\nDBhgbrnlFvP444+X2J5p06aZzp07m5CQEPPggw+aS5cu2c/fuHFj061bN9O5c2fz9NNPF9ovOTnZ\n9OjRwwQGBhqLxWKeeuqpyt2YOgDYb5z8HatPGEopoOalDDl06BBbtmxh7969gHV51cmTJ7N69WqO\nHz/O/v37CQgIcOo6lHO0w1BKFVJTUoY0bNgQi8XCuXPWGfcJCQn84Q9/sEeo16tXj8cee6xc16FK\nV11Tgyilqkh1TxmS79KlS5w6dYp+/axxv8eOHePpp58u9/GV87TDUKoKVPXMrZqcMuTDDz+kW7du\nnDp1it///vf4+vqW65iq4nRISikPq64zt2pKypC+ffty+PBhPv/8c5YvX25Ps9K1a1cOHDhQrnOo\n8tEnjGqmOsULlEXjCSquNgbteTplSIcOHXj22Wd5+eWXefvtt4mPj2fYsGHcc8893HXXXdy4cYPX\nX3+duLi4Sl+bstIOoxrJT0qnv4RVTVHVKUPi4uJYuHAhaWlphIaGsmTJEh566CFycnIQEfuqgco1\nPJ4axJ1qemqQmpYyoqa1VylVXHlSg+g7DKWUUk7RISlV6/n7++Pj44OI4OvrS1JSEr6+vvby/Cmj\n/fr14/r16+zdu5dr165x5swZ+7j7zJkzMcbwwgsv8MUXX/Dpp5/Ss2fxf5SdPXuWMWPGcOHCBUSE\niRMnMmXKFADGjh3Lrl27aNKkCQDjxo3jySef9NBdUKrytMNQdUL+1NAZM2bw4osvsnTp0kLlRaWl\npTF06NBCCx198cUX/O1vf+PRRx8t8TxeXl788Y9/JCwsjOzsbHr06MGgQYMICgoCYMGCBURHR7v4\n6pTyDB2SUnVKRaKY8wUGBpY506d169aEhYUB4OPjQ2BgoD0SWamaTjsMVac4imK2WCxYLBYWL17s\n0nOlpaVx6NAhevfubS+Lj4+3n+/o0aMuPZ9S7qZDUqpaqkw8SkWjmF3phx9+ICoqiiVLltC4cWN7\nuQ5JqZpMOwxV7bgjHsVdHYMjubm5REVFMWrUKIYNG+aRcyrlCdphqGrJYrHUyPgOYwzjx48nMDCQ\np556qqqbo5RLefwdhoh0FpHUAj+XReT3ReoMEJGsAnWe93Q7Vd1Q8B3GmDFjSq37zjvv0K5dOz7+\n+GPuv/9+fvnLXwJw/vx5IiMjAdi7dy+rVq3igw8+sB83OTnZ7dehlCdUaaS3iNQDzgG9jTFfFSgf\nAEwzxpQrrl8jvT3LXe2tafdBqZqsPJHeVT0kdR/wr4KdhaodKvPSWvNpKVU9VfW02t8Cb5ewrY+I\nHBGRf4hI15IOICITRWS/iOzPzMx0TytVueS/tFZK1S5V9oQhIvWBXwPPOdh8EPAzxvwgIpHA34FO\njo5jjHkdeB2sQ1Juaq4qp8q8tHZnindPpgkBWLx4MYmJiYgIISEhrFixggYNGmiaEFUjVeWQ1BDg\noDHmQtENxpjLBT4ni8ifRaSlMeY7j7ZQ1UqeShNy7tw5li5dyvHjx2nYsCExMTGsWbOGsWPHAhqT\noWqeqhySeogShqNExFdsS2yJSC+s7bzowbapOsDdaUIA8vLyuHLlCnl5eeTk5NCmTZsKnU+p6qBK\nOgwRuQUYBPytQFmciOQvjRUNHBORw8BS4LemNi3coaoFd6cJadu2LdOmTcPPz4/WrVvTpEkTIiIi\n7Ns1TYiqaapkSMoY8yPQokjZsgKfXwVe9XS7VPWRmprqkncZVZkm5NKlS2zatIkzZ87QtGlThg8f\nzurVq3n44YcBHZJSNU9VT6tVqpidO3e69cW3p9KEbN++nQ4dOtCqVSsAhg0bxkcffWTvMJSqabTD\nUNVSbQja8/PzY9++feTk5NCwYUNSUlJKnE2lVE1Q1XEYSlUbrk4T0rt3b6KjowkLCyMkJIQbN24w\nceJEt1+HUu5SpalBXE1Tg3hWSe2tadehVF1Wk1KDqBrO0ctpTe2hVO2kQ1Kqwnbu3Kkdg1J1iD5h\nqEpxNOzkzhlO5eXJVCBnz55lzJgxXLhwARFh4sSJTJkyBaBQKhBjDIsWLeK+++4DrAsuzZo1i40b\nN+Lj44O3tzfPP/88Q4YM8dBdUso52mGoWs9TqUC8vLz44x//SFhYGNnZ2fTo0YNBgwYRFBQE/Dfu\nYseOHUycOJFTp04BMGvWLDIyMjh27Bje3t5cuHCBXbt2ufIWKOUS2mGoOqNfv372zqK8AgMDy6zT\nunVrWrduDYCPjw+BgYGcO3fO3mHkCw8P59y5cwDk5OTwxhtvcObMGby9vQG4/fbbiYmJqVA7lXIn\n7TBUtVbe4a3SZmY5SgWSPyQVGxvL1KlTK9JEh9LS0jh06BC9e/cutm3r1q08+OCDAJw+fRo/Pz8a\nN27ssnMr5S7aYahqK39djcq+WPdUKpB8P/zwA1FRUSxZsqRQRxAfH8+MGTNIT0/n448/dvl5lXI3\n7TBUtVaZdTXyeSoVCFhfYEdFRTFq1CiGDRtWaFv+O4xXXnmFcePGceDAAQICAvj666+5fPmyPmWo\nak+n1SrlIsYYxo8fT2BgIE899VSJ9SZPnsyNGzfYtm0bjRo1Yvz48UyZMoVr164BkJmZyfr16z3V\nbKWcph2GqrNcnQpk7969rFq1ig8++MB+3OTk5GLHEhFmzpxJQkICAHPnzqVVq1YEBQURHBzM0KFD\n9WlDVUuaGqQaqS0pNVx1HbXlfihVnWlqEFXlXLGehaYYUap60SEp5XLVOWWIv78/ISEhhIaGEhER\nwTfffFOoPH8o6cknn+Txxx/HYrEQFBREw4YN7ds2bNjA+vXr6dq1KzfddBOlPdUuXryYrl27Ehwc\nzEMPPcTVq1cBa+R3hw4dsFgsdOvWjZSUFPs+ubm5PPvss3Tq1ImwsDDCw8P5xz/+4d4bo5QzjDFO\n/QC3AX75P07UfxP4FjhWoKw58D5wyvZnsxL2HQx8CZwGnnW2jT169DA1Wf/+/U3//v2ruhnVhjvu\nxx133GEyMzONMcY899xz5oknnihWXtSZM2dM165dC5UdP37cnDhxwvTv39989tlnDvdLT083/v7+\nJicnxxhjzPDhw82KFSuMMcbExsaa9evXG2OM+eCDD0xAQIB9v+nTp5sxY8aYq1evGmOM+eabb8za\ntWsreMVKlQ7Yb5z8HVvmE4aI/FpETgFngF1AGuDMP3dW2n7xF/QskGKM6QSk2L4XPV894DVgCBAE\nPCQiQUXrKVVZ/fr14/Tp0xXaNzAw0J5rqjR5eXlcuXKFvLw8cnJyaNOmTbE6jiK/X3nlFY38VtWO\nM0NS/wvcDZw0xnQA7gP2lbWTMWY38J8ixQ8A/2f7/H/Agw527QWcNsb82xhzDVhj208pl3IU+Z0/\n7LR48eJKH79t27ZMmzYNPz8/WrduTZMmTYiIiChWTyO/VU3hzEvvXGPMRRG5SURuMsbsEJElFTzf\n7caYDNvnb4DbHdRpC5wt8D0dKJ5fQdUJlX157miGlacivy9dusSmTZs4c+YMTZs2Zfjw4axevdq+\nprdGfquaxpkO43sRuRXYDbwlIt8CP1b2xMYYIyKVntMrIhOBiWBdQ7kiqks6bp0VVNjOnTvd8t/G\nU5Hf27dvp0OHDrRq1QqAYcOG8dFHH9k7DI38VjWNMx3GA8BVYCowCmgCzKng+S6ISGtjTIaItMb6\nUryoc0D7At/b2cocMsa8DrwO1jiM8jbIVfmKlHvU5BgMPz8/9u3bR05ODg0bNiQlJcXhOhqTJ0/m\nzTffZNu2bfzyl7+0R37/5S9/oX79+mRmZrJz506GDx9eBVeh1H8502HcYYw5bvv8fwAiMgDYWYHz\nvQvEAi/Z/tzkoM5nQCcR6YC1o/gtMLIC53KaK/IVuUJ1edKpqwpmrw0NDSUpKanEuu+88w5PPPEE\nmZmZ3H///VgsFrZt28b58+eZMGECycnJ9O7dm+joaMLCwvDy8qJ79+5MnDix2LEKRn7/8pe/ZO7c\nucycOZOgoCAaNGjALbfcwpw5Ff03mlKuU2akt4gcA1YBCUAD2589jTHhZez3NjAAaAlcAP4H+Duw\nDuvU3K+AGGPMf0SkDZBojIm07RsJLAHqAW8aY+Y5czEVifSuTtHE1aktSqm6wdWR3r2Bl4GPAB/g\nLeDnZe1kjHmohE33Oah7Hogs8D0ZKJ6ERymlVJVxZlptLnAFaIj1CeOMMeaGW1ullFKq2nGmw/gM\na4fxM6Av1kA6zb2sysXf35+oqCj79w0bNjB27NgKHeutt94iNDSUkJAQ+vTpw+HDhx3WGzVqFJ07\ndyY4OJhx48aRm5sLwMqVK2nVqpXTmWqVUlbOdBjjjTHPG2NyjTEZxpgHsL68VqpcDhw4wPHjx8uu\nWIYOHTqwa9cujh49yqxZsxy+SAZrh3HixAmOHj3KlStXSExMtG8bMWIEqamppKamlvpyWyn1X2V2\nGMaY/QAicpuI+ImIH9YUIUqVy9NPP828ecXnL3z66aeEh4fTvXt3+vTpw5dffgnA3Xffzeeff26v\nN2DAAPbv30+fPn1o1qyZvU56errD80VGRiIiiAi9evUqsZ5SyjnO5JL6VQVzSak6qqTpwTExMRw8\neLBY/qYuXbrw4YcfcujQIebMmcOMGTMA61PAunXrAMjIyCAjI6NYHMPy5csZMmRIqe3Jzc1l1apV\nDB7839Rma9eutQ9JrVixoryXqFSd5MwsqblYc0ltN8Z0F5GBwMPubZaqqfIDIR2pV68e8fHxzJ8/\nv9Av+aysLGJjYzl16hQiYn/XEBMTQ0REBLNnz2bdunVER0cXOt6OHTtYvnw5e/bsKbVNkyZNol+/\nfvTt29deNmLECF599dWKXqZSdZJTs6SMMRcBey4pwKk5u6puKi1qfvTo0ezevZuzZ/+bLmzWrFkM\nHDiQY8eOsXnzZvuaEW3btqVFixYcOXKEtWvXMmLECPs+R44cYcKECWzatIkWLVqUeL7Zs2eTmZnJ\nokWLXHBlStVtznQYRXNJ/Qn4wb3NUrXVzTffzNSpUwtlg83KyqJt27aAdQZTQSNGjCAhIYGsrCxC\nQ0MB+Prrrxk2bBirVq3irrvuKvFciYmJbNu2jbfffpubbtK1wpSqLGf+LzoM5GDNJbUV+Bdwwp2N\nUrXb+PHjycvLs39/5plneO655+jevXuhcoDo6GjWrFlTaD2IOXPmcPHiRSZNmoTFYin0XiMyMpLz\n588DEBcXx4ULFwgPD8disWh6DaUqyZnUIAeNMWFFyo4YY0Ld2rIK0NQgVa82XINSdYlLUoOIyGPA\nJOBOETlSYJMPsLdyTVS1RdEZUZr5V6naq7RZUn/FOn12PoWXUs02xhRdSU/VQe5KDe/v74+Pjw8i\ngq+vL0lJSfj6+trL8zPK9uvXj+vXr7N3716uXbvGmTNn7Mumzpw5s9isqrLExsaSnJxM27ZtS5zp\ndenSJR5++GHS09PJy8tj+vTpjBkzhry8PLy9vQut4Ld582bat2/v8DhK1UQldhjGmCwgCygpiaBS\nxVLDuypFe/4iRzNmzODFF19k6dKlhcqLSktLY+jQoSX+onfGuHHjePzxx0uMHAd45ZVXsFgsvPfe\ne1y4cIHkYNTbAAAgAElEQVQuXbowcqQ1+76Pj0+lzq9UdadTR1S11q9fv2KBfs5avHgxQUFBhIaG\n2le5mzlzJuPHj6d///507NiR1157zV6/f//+NG/evNRjigjZ2dkA/PDDD7Rs2dL+xKNUbedM4J5S\nVWbLli2FhnkKLnIUGxvL1KlTS9w3ISGBr776ivr16/P999/by0+ePElKSgrff/89gYGBxMXFOf1L\nf8qUKQwdOpQ2bdpw+fJlNmzYgIgAkJ2dbR+eCwgIYMOGDeW+XqWqM+0wlMulpqaWa2jK0Yyq/I4h\nNDSUuXPn2svLsx53165defjhh3nggQd48MEH7eVDhw6lfv363HbbbTRv3pzMzEx8fX2dOmZycjK9\nevVi165dnDx5ksGDB3P06FEaNGigQ1Kq1tMhKeVSO3fudMlL8B07dtgzyTZt2rRCx9i2bRtxcXF8\n9tln9OrVi+vXrwPg7e1tr1OvXr1isR+lWbFiBVFRUYgInTt3pn379pw8ebJC7VOqptEnDOVy1SEG\n4/r166Snp3Pvvfdyzz330L59e3Jycip9XD8/P1JSUggPDycjI4PTp0/ToUMHF7RYqerPbR2GiLwJ\nDAW+NcYE28oWAL8CrmGNGH/EGPO9g33TgGzgOpDnbFCJcr+CQ01VEXNR8B1GaGhoiWtZ5OXlMXLk\nSLKzs7lx4wbTpk3Dx8en1GMPHz6cPXv2cPHiRdq1a8fcuXMZO3Ysr732Gt7e3kyYMIEXXniBsWPH\nsm7dOm7cuMHChQtp1qxZuZ5SlKqpyoz0rvCBRfphzTmVVKDDiAA+MMbkicjLAMaY6Q72TQN6GmO+\nK885NdLbvYrGXeR/rq7tVUqVzSWR3pVljNktIv5Fyv5Z4Os+oHyRVarKFewgXBVzoZSqGarypfc4\nSl6IyQDbReSAiJQcRQWIyEQR2S8i+zMzM13eSKWUUlZV8tJbRP4A5AFvlVDlHmPMORG5DXhfRE4Y\nY3Y7qmiMeR14HaxDUm5psCo3V6X3+OSTT9i8eTP169fnzjvvZMWKFQ5nTS1evJjExEREhJCQEFas\nWEGDBg0YO3Ysu3btokmTJoA1mvvJJ5/03I1QqhbxeIchImOxvgy/z5TwAsUYc87257ci8g7QC+t6\nHKoGcUV6j8aNGzN//ny8vLyYPn068+fP5+WXXy6037lz51i6dCnHjx+nYcOGxMTEsGbNGsaOHQvA\nggULyp1XSilVnEeHpERkMPAM8GtjjMM5jiJyi4j45H8GIoBjnmulcrXKpPeIiIjAy8v675q7776b\n9PR0h/Xy8vK4cuUKeXl55OTk0KZNmwq3VynlmNs6DBF5G/gY6Cwi6SIyHngVa3r090UkVUSW2eq2\nEZFk2663A3tE5DDwKfCeMWaru9qpKseZqG5H6T0sFgsWi6XQyntlefPNNwutBZ6vbdu2TJs2DT8/\nP1q3bk2TJk2IiIiwb4+Pj7ef7+jRo06fTylVmDtnSTnKcru8hLrngUjb538D3dzVroLclZ67rti5\nc2epnYUr0nvkmzdvHl5eXowaNarYtkuXLrFp0ybOnDlD06ZNGT58OKtXr7YnHNQhKaVco85Geud3\nFqpkziyOVFoMRkU6BkdWrlzJli1bSElJsSf6K2j79u106NCBVq1aATBs2DA++ugje4ehlHKNOp1L\nKn+YQhVXXTrUrVu3kpCQwLvvvkujRo0c1vHz82Pfvn3k5ORgjCElJYXAwEAPt1Sp2q/OPmGosrlr\ncSRn03sATJ48mZ9++olBgwYB1hffy5Yt4/z580yYMIHk5GR69+5NdHQ0YWFheHl50b1791IXQVJK\nVYx2GMot0tLSylUO1tiNY8cKT4graXZVmzZtSE5Otn+fPXs2s2fPLlZv5cqVZbZVKeWcOj0kpZRS\nynn6hKEqxZMR3WfPnmXMmDFcuHABEWHixIlMmTIFQCO6lfIA7TBUpXkqotvLy4s//vGPhIWFkZ2d\nTY8ePRg0aBBBQUGATp9Vyt10SEq5jLsjulu3bk1YWBgAPj4+BAYGcu7cuYo3WClVLtphKJdxd0R3\nQWlpaRw6dIjevXvbyzSiWyn30iEpVWmeiujO98MPPxAVFcWSJUto3LixvVyHpJRyL+0wqhlncjN5\nqh3ORnV7KqIbIDc3l6ioKEaNGsWwYcMqfU6llPN0SKoa2blzZ52NPHcmotsYw/jx4wkMDOSpp57y\ncAuVUvqEUc1Ul/WxXfGU4+qI7r1797Jq1SpCQkLsHeuLL75IZGRkpduqlCqblLCGUY3Us2dPs3//\nfqfqFv2FWF1+UVcX+fdH74tStZuIHDDG9HSmrg5JKaWUcop2GEoppZyi7zCUU1yVAqS8015jY2NJ\nTk6mbdu2JaZbT0pKYsGCBYgIPj4+LFu2jJCQEPLy8vD29i4UG7J582bat29fwbugVN3mtg5DRN4E\nhgLfGmOCbWUvAL8DMm3VZhhjkh3sOxj4E1APSDTGvOSudirnuSIFSHmNGzeOxx9/vNR05XfeeScf\nfvghTZs2ZfPmzcTFxbF3717AGhFeHdb1UKo2cOeQ1EpgsIPyxcYYi+3HUWdRD3gNGAIEAQ+JSJAb\n26nKqTIpQBYvXkxQUBChoaH2FfFmzpzJ+PHj6d+/Px07duS1116z1+/fvz/Nmzcv9Zg///nP7YkK\nS0oropSqPHeu6b1bRPwrsGsv4LRtbW9EZA3wAHDcda1TleEoBUj+kFRsbCxTp04tcd+EhAS++uor\n6tevz/fff28vP3nyJCkpKXz//fcEBgYSFxdnP2Z5LF++vFBakezsbPsU3ICAADZs2FDuYyqlrKri\nHcYTIjIG2A88bYy5VGR7W+Bsge/pQG+URzmK9HZFCpCuXbvy8MMP88ADD/Dggw/ay4cOHUr9+vW5\n7bbbaN68OZmZmfj6+parzdu3b2fVqlXs2bPHXqZDUkq5jqdnSf0/oCNgATKAP1b2gCIyUUT2i8j+\nzMzMsndQTim6PCtYO4bU1FSSkpKKrVXhrG3bthEXF8dnn31Gr169uH79OgDe3t72OvXq1SMvL69c\nx01NTeXRRx9l06ZNNGvWrEJtU0qVzqNPGMaYC/mfReQNYIuDaueAgtNY2tnKSjrm68DrYA3cc01L\nlTsC9q5fv056ejr33nsv99xzD+3btycnJ6fSx01LSyM6Opq//vWvBAQEuKClSilHPNphiEhrY0yG\n7etvgGMOqn0GdBKRDlg7it8CIz3URFUBzqYAycvLY+TIkWRnZ3Pjxg2mTZuGj49PqccePnw4e/bs\n4eLFi7Rr1465c+cyduxYXnvtNby9vZkwYQIvvPAC//nPf3j00UcB69PKJ5984tqLVEq5LzWIiLwN\nDABaAheA/7F9twAGSAMeNcZkiEgbrNNnI237RgJLsE6rfdMYM8+Zc2pqEKWUKp/ypAZx5yyphxwU\nLy+h7nkgssD3ZKDYlFullFJVRyO9HXBVVPMnn3zC5s2bqV+/PnfeeScrVqxw+LJ48eLFJCYmIiKE\nhISwYsUKGjRowNixY9m1axdNmjTBGMOiRYu47777AOu6ELNmzWLjxo34+Pjg7e3N888/X+ZKdUop\nVVHaYZTAFVHNjRs3Zv78+Xh5eTF9+nTmz5/Pyy+/XGi/c+fOsXTpUo4fP07Dhg2JiYlhzZo1jB07\nFvjvKnI7duxg4sSJnDp1CoBZs2aRkZHBsWPH8Pb25sKFC+zatctNd0MppTT5YJkqE9UcERGBl5e1\nTy4tAjkvL48rV66Ql5dHTk4Obdq0KVYnPDycc+esk8VycnJ44403eOWVV+zTUW+//XZiYmIq1E6l\nlHKGdhhlcBTVbLFYsFgsLF682OnjvPnmmw6Hi9q2bcu0adPw8/OjdevWNGnShIiIiGL1tm7dag90\nO336NH5+foXWs1ZKKXfTIakSuCKqOd+8efPw8vJi1KhRxbZdunSJTZs2cebMGZo2bcrw4cNZvXq1\nPc9SfHw8M2bMID09nY8//rhyF6WUUpVQ5zuMkqbTVqRjcGTlypVs2bKFlJQURKTY9u3bt9OhQwda\ntWoFwLBhw/joo4/sHUb+O4xXXnmFcePGceDAAQICAvj666+5fPmyPmUopTxGh6TcaOvWrSQkJPDu\nu+/SqFEjh3X8/PzYt28fOTk5GGNISUkhMDCwWL3Jkydz48YNtm3bRqNGjRg/fjxTpkzh2rVrAGRm\nZrJ+/Xq3Xo9Sqm7TDqOcCr7DGDNmTKl1J0+eTHZ2NoMGDcJisRAXFwfA+fPniYy0hp307t2b6Oho\nwsLCCAkJ4caNGw7XfhARZs6cSUJCAgBz586lVatWBAUFERwczNChQ/VpQynlVm6L9K4KFYn01ghv\npVRdVp5Ib33CUEop5RTtMJRSSjlFOwwn+Pv7ExUVZf++YcMGeyR2eb311luEhoYSEhJCnz59OHz4\nsMN6o0aNonPnzgQHBzNu3Dhyc3MB66yrVq1aYbFY6NKlS7FYkKSkJIKDgwkJCaF79+4sXLiwQu1U\nSqmitMNw0oEDBzh+vPKrxHbo0IFdu3Zx9OhRZs2a5fAFN1g7jBMnTnD06FGuXLlCYmKifduIESNI\nTU1l7969zJs3j7NnrQsU/uMf/2DJkiX885//5OjRo+zbt48mTZpUus1KKQXaYTjt6aefZt684lnW\nP/30U8LDw+nevTt9+vThyy+/BKypQD7//HN7vQEDBrB//3769OljXxGutHQhkZGRiAgiQq9evRzW\na9GiBQEBAWRkWJcYmT9/PgsXLrSnFvH29uZ3v/td5S5cKaVstMNwUkxMDAcPHiyWV6pLly58+OGH\nHDp0iDlz5jBjxgzA+hSwbt06ADIyMsjIyKBnz8ITEZYvX15mdtnc3FxWrVrF4MGDi237+uuvuXr1\nKqGhoQAcO3aMHj16VPgalVKqNNphFFF0YaV89erVIz4+nvnz5xcqz8rKYvjw4QQHBzN16lT7U0VM\nTAwbNmwAYN26dURHRxfab8eOHSxfvrxY9tqiJk2aRL9+/ejbt6+9bO3atYSGhhIQEMCkSZNo0KBB\neS9TKaXKTTuMAgYMGFAoPXlRo0ePZvfu3fZ3BmBNMz5w4ECOHTvG5s2buXr1KmBNKtiiRQuOHDnC\n2rVrGTFihH2fI0eOMGHCBDZt2kSLFi1KPN/s2bPJzMxk0aJFhcpHjBjBkSNH+Oijj3j22Wf55ptv\nAOjatSsHDhyo0LUrpVRZ3NZhiMibIvKtiBwrULZWRFJtP2ki4vC3s23bUVs95yLxXMRisZS47eab\nb2bq1KmFZiZlZWXRtm1bwDqDqaARI0aQkJBAVlaWfdjo66+/ZtiwYaxatYq77rqrxHMlJiaybds2\n3n77bW66yfF/pp49ezJ69Gj+9Kc/AfDcc88RHx9v70CuXbtW6GW5UkpVhjufMFYChQbejTEjjDEW\nY4wF2Aj8rZT9B9rqOhWB6Cnjx48nLy/P/v2ZZ57hueeeo3v37oXKAaKjo1mzZk2hdSrmzJnDxYsX\nmTRpEhaLpdB7jcjISM6fPw9AXFwcFy5cIDw8HIvFwpw5cxy2Z/r06axYsYLs7GwiIyOZPHkyv/jF\nL+jatSthYWFcvnzZlZevlKrD3JoaRET8gS3GmOAi5QJ8DdxrjDnlYL80oKcx5rvynK+yqUE0XYhS\nqq6pCalB+gIXHHUWNgbYLiIHRMRxoIJSSimPqqr1MB4C3i5l+z3GmHMichvwvoicMMbsdlTR1qFM\nBGuqcKWUUu7h8Q5DRLyAYUCJAQPGmHO2P78VkXeAXoDDDsMY8zrwOliHpFzdXn9/f3x8fBARfH19\nSUpKwtfX115er149wLr29/Xr19m7dy/Xrl3jzJkzdO7cGYCZM2fyySefsHnzZurXr8+dd97JihUr\naNq0aaFznT17ljFjxnDhwgVEhIkTJzJlyhQAxo4dy65du2jSpAnGGBYtWsR9990HWGM1Zs2axcaN\nG/Hx8cHb25vnn3++zBgPpZQqF2OM234Af+BYkbLBwK5S9rkF8Cnw+SNgsDPn69Gjh3FW//79TZMm\nTUz//v3tP/nfC7rjjjtMZmamMcaY5557zjzxxBPFyos6c+aM6dq1a6Gybdu2mdzcXGOMMc8884x5\n5plniu13/vx5c+DAAWOMMZcvXzadOnUyn3/+uTHGmNjYWLN+/XpjjDEffPCBCQgIsO83ffp0M2bM\nGHP16lVjjDHffPONWbt2rdP3QilVdwH7jZO/0905rfZt4GOgs4iki8h426bfUmQ4SkTaiEiy7evt\nwB4ROQx8CrxnjNnqrnaWR79+/YpFejsrIiICLy/rA11JKUFat25NWFgYAD4+PgQGBnLu3Lli9cLD\nw+3lOTk5vPHGG7zyyit4e3sDcPvttxeamaWUUq7gtiEpY8xDJZSPdVB2Hoi0ff430M1d7SrIYrEU\nmyWVmppa4mypLVu2EBISYv8+cOBA+5BUbGwsU6dOdeq8b775ZqFAPkfS0tI4dOgQvXv3LrZt69at\nPPjggwCcPn0aPz8/XW1PKeV2VfXSu9or2HHkdwyhoaHMnTvXXmfHjh20bNmyXMedN28eXl5ejBo1\nqsQ6P/zwA1FRUSxZsqRQRxAfH8+MGTNIT0/n448/Lt8FKaVUJWmHUUT+U0fBnFIV6RgcWblyJVu2\nbCElJQVrKEpxubm5REVFMWrUKIYNG1Zo24IFC4iOjuaVV15h3LhxHDhwgICAAL7++msuX76sTxlK\nKbfSXFIesnXrVhISEnj33Xdp1KiRwzrGGMaPH09gYCBPPfVUiceaPHkyN27cYNu2bTRq1Ijx48cz\nZcoUrl27BkBmZibr1693y3Uopeou7TAqYeDAgVgsFiwWC2PGjCm17uTJk8nOzmbQoEFYLBbi4uIA\nOH/+PJGRkQDs3buXVatW8cEHH9iPm5ycXOxYIsLMmTNJSEgAYO7cubRq1YqgoCCCg4MZOnSoPm0o\npVzOralBPM2VqUE0TYhSqi6oCalBqq38l92lpTlXSqm6SDuMAnbu3FlqenN38ff3JyQkhNDQUCIi\nIuzpyfPL84ennnzySR5//HEsFgtBQUE0bNjQvi1/sabyiI2NpVWrVqVe86VLl7j//vvp1q0bXbt2\nJSkpCYC8vDzq1auHxWIhODiYX//614Uy4544cYIhQ4bQqVMnwsLC+O1vf8u3335b7jYqpaoRZyP8\nasJPeSO9i0Z1l2e7K7kqmry8du7caT755BPTrVu3EuvMnj3bzJgxwxhjjSBv2rSpyc3NNbm5uaZJ\nkyb2eiNHjjQvvfSSMcaYH3/80XTs2NG899579u3bt283x48fr1R7lVKuR3WI9FYVU5lo8sWLFxMU\nFERoaCgPP/wwYM1jNX78ePr370/Hjh157bXX7PX79+9P8+bNSz2miJCdnQ1Y40NatmxpD1YsqGD0\n+erVq+nfv7/9ZT7AfffdR2BgYIWuSylVPWgcRjVTmWjyhIQEvvrqK+rXr8/3339vLz958iQpKSl8\n//33BAYGEhcX5/CXviNTpkxh6NChtGnThsuXL7Nhw4ZiMSTXr1/ngw8+YNKkSQAcO3aMHj1KzC2p\nlKqhtMMoRcFob1dyNPPKFdHkXbt25eGHH+aBBx6wpw4BGDp0KPXr1+e2226jefPmZGZm4uvr69Qx\nk5OT6dWrF7t27eLkyZMMHjyYo0eP0qBBA7Kzs7FYLKSnpxMSEsLAgQOdOqZSqmbSIakSePoF+I4d\nO0hNTSUpKalY2nNnbdu2jbi4OD777DN69erF9evXAexJCQHq1atXbCnZ0qxYsYKoqChEhM6dO9O+\nfXtOnjwJWBMkpqam8tVXX/HTTz/xl7/8BbB2XAcOHKjQNSilqi99wihFTYrBuH79Ounp6dx7773c\nc889tG/fnpycnEof18/Pj5SUFMLDw8nIyOD06dN06NChUJ1bbrmFP/3pT8TExPDoo48yevRoXn75\nZbZu3crgwdZl3Xfs2IGvr6++x1CqBtMnjGrO2WjyvLw8Ro4cSWhoKGFhYUybNg0fH59Sjz18+HD6\n9u3L8ePHadeuHStXrgTgtddeIzExEYAXXniBXbt2ERoayqBBg1i4cCHNmjUrdqyf/exndOnShXXr\n1tGoUSO2bNnC4sWL6dSpE0FBQbzxxhsuycellKo6Guldg54ilFLK1TTSWymllMtph6GUUsop+tLb\nxt/fHx8fH0QEX19fkpKS8PX1tZfnxy3069eP69evs3fvXq5du8aZM2fo3LkzYA2Si46OLtd5Y2Nj\nSU5Opm3btiXmr0pKSmLBggWICD4+PixbtoyQkBDy8vLw9va2f+7YsSOrV6+2Z6o9ceIEU6dO5fTp\n0/j4+HDXXXexdOlSbrvttkrcKaVUneVsSHh5f4D2wA7gOPA5MMVW3hx4Hzhl+7NZCfsPBr4ETgPP\nOnPOyqQGqc7pOfbs2WMuXbpkjDHm3XffNX369DHGGE3PoZSqNKpJapA84GljTBBwN/C4iAQBzwIp\nxphOQIrteyEiUg94DRgCBAEP2fb1iOqWnuPnP/+5PTbj7rvvJj093WE9Tc+hlHIntw1JGWMygAzb\n52wR+QJoCzwADLBV+z9gJzC9yO69gNPGmH8DiMga237HXdW+1NTUEgPzqlt6joKWL1/OkCFDipVr\neg6llLt55B2GiPgD3YFPgNttnQnAN8DtDnZpC5wt8D0d6F3CsScCE8EaZOas/LW7C6qu6Tnybd++\nnVWrVrFnzx57mabnUEp5ittnSYnIrcBG4PfGmMsFt9nGzyoVCGKMed0Y09MY07NVq1ZO7+co/qK6\npucA6xPRo48+yqZNmwoFzml6DqWUp7i1wxCRm7F2Fm8ZY/5mK74gIq1t21sDjlbVOYf1pXm+dray\naq1geo6EhAS+++47l6TnSEtLIzo6mr/+9a8EBAQ4rJOfnmPBggVcv36d0aNHs3PnTrZu3Wqvs2PH\nDr744otKt0cpVTe5rcMQaw7s5cAXxphFBTa9C8TaPscCmxzs/hnQSUQ6iEh94Le2/apEdUjP8Z//\n/IdHH30Ui8VC794OR+c0PYdSyq3clhpERO4BPgSOAjdsxTOwvsdYB/gBXwExxpj/iEgbINEYE2nb\nPxJYAtQD3jTGzCvrnOVJDaKUUqp8qUHcOUtqDyAlbL7PQf3zQGSB78lAsntap5RSqrw0NYhSSimn\naIehlFLKKdphKKWUckqtWg9DRDKxvkh3VkvgOzc1pybT+1Kc3pPi9J4UVxPvyR3GGKeC2GpVh1Fe\nIrLf2dkBdYnel+L0nhSn96S42n5PdEhKKaWUU7TDUEop5ZS63mG8XtUNqKb0vhSn96Q4vSfF1ep7\nUqffYSillHJeXX/CUEop5aQ60WGIyGAR+VJETouIoxX+RESW2rYfEZGwqminJzlxT7qIyMci8pOI\nTKuKNlYFJ+7LKNvfkaMi8pGIdKuKdnqSE/fkAds9SRWR/bY8crVaWfekQL2fiUieiER7sn1u4+xa\nrjX1B2vywn8BHYH6wGEgqEidSOAfWHNf3Q18UtXtrgb35DbgZ8A8YFpVt7ka3Zc+2Nahx7qEsP5d\ngVv57/B2KHCiqttd1fekQL0PsObEi67qdrvipy48YdiXezXGXAPyl3st6AEgyVjtA5rmr9lRS5V5\nT4wx3xpjPgNyq6KBVcSZ+/KRMeaS7es+rGu11GbO3JMfjO03JHALlVwUrQZw5ncKwBNY1wNytOZP\njVQXOgxHy722rUCd2qSuXa+zyntfxmN9Mq3NnLonIvIbETkBvAeM81DbqkqZ90RE2gK/Af6fB9vl\ndnWhw1DK5URkINYOY3pVt6U6MMa8Y4zpAjwI/G9Vt6caWAJMN8bcKLNmDeK29TCqEWeWe62RS8JW\nQl27Xmc5dV9EJBRIBIYYYy56qG1VpVx/V4wxu0Wko4i0NMbUtJxKznLmnvQE1lgXHqUlECkiecaY\nv3umie5RF54wnFnu9V1gjG221N1AljEmw9MN9aBqtQRuNVLmfRERP+BvwGhjzMkqaKOnOXNPAmxL\nMmObYegN1OaOtMx7YozpYIzxN8b4AxuASTW9s4A68IRhjMkTkcnANv673OvnIhJn274M6yyGSOA0\nkAM8UlXt9QRn7omI+AL7gcbADRH5PdaZIJerrOFu5uTfleeBFsCfbb8j80wtTjbn5D2JwvoPrlzg\nCjCiwEvwWsfJe1IraaS3Ukopp9SFISmllFIuoB2GUkopp2iHoZRSyinaYSillHKKdhhKKaWcoh2G\nqrVExF9Ejnn4nDtFpNRptu5ql4gMEJE+Bb6vrDVZUlW1oB2GqvNEpF5Vt8FFBmDNpquUW2iHoWo7\nLxF5S0S+EJENItIIQETSRORlETkIDBeR34nIZyJyWEQ2Fqi30rZWykci8u+C/2IXkem2dTEOi8hL\nBc45XEQ+FZGTItK3tMaJSD0RWWA79xERedRWPsD2tLJBRE7YriE/mjrSVnbA1rYtIuIPxAFTbetS\n5J+3n6O2K1UR2mGo2q4z8GdjTCBwGZhUYNtFY0yYMWYN8DdjzM+MMd2AL7AmFszXGrgHGAq8BCAi\nQ7CmtO5t2yehQH0vY0wv4PfA/5TRvvFYU9H8DOv6I78TkQ62bd1txwjCuvbCz0WkAfAXrHmsegCt\nAIwxacAyYLExxmKM+bCktitVUdphqNrurDFmr+3zaqy/PPOtLfA5WEQ+FJGjwCiga4FtfzfG3DDG\nHAdut5X9AlhhjMkBMMb8p0D9v9n+PAD4l9G+CKxpNVKBT7CmHelk2/apMSbdlvE01XasLsC/jTFn\nbHXeLuP4jtquVIXU+lxSqs4rmvum4PcfC3xeCTxojDksImOxvg/I91OBz+LEOfPrX6fs/8cEeMIY\ns61QociAIud15liltSX/XEpVmD5hqNrOT0TCbZ9HAntKqOcDZIjIzVifMMryPvBIgXcdzSvYvm3A\nY7bzIiJ3icgtpdT/Euhoe2cBMKLAtmys16GUW2iHoWq7L4HHReQLoBklr4A2C+uQ0F7gRFkHNcZs\nxfWmsR8AAACTSURBVJrSer9tOGlaBduXCBwHDtqm2v6FUp4kjDFXsL6H2SoiB7B2Elm2zZuB3xR5\n6a2Uy2i2WqVqGBG51Rjzg23W1GvAKWPM4qpul6r99AlDqZrnd7anms+BJlifSpRyO33CUEop5RR9\nwlBKKeUU7TCUUko5RTsMpZRSTtEOQymllFO0w1BKKeUU7TCUUko55f8DaRicD5/VNBsAAAAASUVO\nRK5CYII=\n",
      "text/plain": [
       "<matplotlib.figure.Figure at 0x151132ce48>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "tree.rooted = True\n",
    "Phylo.draw(tree)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "?Phylo.draw"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
